jlm/llvm/opt/InvariantValueRedirection.hpp

```diff
@@ -34,6 +37,13 @@ class RvsdgModule;
  * The output of a gamma node is considered invariant if all the corresponding region results are
  * connected to the arguments of a single gamma input. All the users of a theta output are diverted
```

Comment:
You mean changing "gamma input" to "gamma output"? If so, then this would be wrong. We check whether a gamma output is invariant by looking at all the subregion results of the gamma output, if all of them are connected to a subregion argument and the respective input(s) of these arguments have the same origin, then it is invariant. See the [IsInvariant() method](https://github.com/phate/jlm/blob/21618b0d4a3381d22deedde541ea3d7ab6f135cb/jlm/rvsdg/gamma.cpp#L279C1-L304C2) for a concrete implementation.

Reply:
No, I think "theta output" should be "gamma output"

Reply:
Ah, I missed that part. Yes, I will fix it. Nicely spotted!

Reply:
Fixed.